tt/problem_history_edit.html.tt

```diff
@@ -49,6 +49,10 @@
   $('#wrap_button').click(function() { editor.execCommand('toggleWrapMode'); });
   $('#keyword_button').click(add_keyword);
   $('.problem_menu_trigger').click(function() { $(this).toggleClass('problem_menu_hovered'); });
+  $('#edit_file_form').submit(function() {
+    $('<input/>').attr({ type: 'hidden', name: 'new_name', value: $('#name_input').val() }).appendTo('#edit_file_form');
```

Comment:
Add this `input` to template with empty value and just set the value here.
Also, I dislike that there is no fallback without JS. I suggest to use `form` attribute by default, with JS as fallback for IE.